pages/docs/tutorials/native/apple-framework.md

```diff
@@ -0,0 +1,284 @@
+---
+type: tutorial
+layout: tutorial
+title:  "Kotlin/Native as an Apple Framework"
+description: "Compiling Kotlin/Native code and use it from Objective-C and Swift"
+authors: Eugene Petrenko
+date: 2018-08-05
+showAuthorInfo: false
+issue: EVAN-5132
+---
+
+Kotlin/Native provides bidirectional interoperability with Objective-C/Swift. 
+Objective-C frameworks and libraries can be used in Kotlin code.
+Kotlin module can be used in Swift/Objective-C code too.
+
+Besides that, Kotlin/Native has C interop. You may want to
+see the [Kotlin/Native as a Dynamic Library](dynamic-libraries.html)
+tutorial too.
+
+In that tutorial, we will see how to use Kotlin/Native code from
+Objective-C and Swift applications on macOS and iOS.
+We will build a Framework from a Kotlin code.
+
+In the tutorial we'll: 
+- [Create a Kotlin Library](#creating-a-kotlin-library) and compile it to Apple Framework
+- Examine generated [Objective-C and Swift API](#generated-framework-headers)
+- Use the Framework from [Objective-C](#using-the-code-from-objective-c) and [Swift](#using-the-code-from-swift)
+
+   
+## Creating a Kotlin Library
+
+Kotlin/Native compiler is able to produce a Framework for macOS and iOS
+out of the Kotlin code. The produced framework contains all declarations
+and binaries to use it with Objective-C and Swift
+
+The best way to understand the techniques is to try those techniques. 
+Let's create a tiny Kotlin library first and use it from an Objective-C program.
+
+We create a `lib.kt` file with the library contents:
+
+<div class="sample" markdown="1" mode="kotlin" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```kotlin
+package demo
+
+object DemoObject {
+  val field = "A"
+}
+
+class DemoClass {
+  fun foo() : Long = 42
+}
+
+fun ints(b: Byte, s: Short, i: Int, l:Long) { }
+fun floats(f: Float, d: Double) { }
+
+fun strings(str: String) : String {
+  return "That is '$str' from C"
+}
+
+val globalString = "A global String"
+```
+</div>
+
+We need to have a Kotlin/Native compiler on our machines. 
+You may have a look at the
+[A Basic Kotlin/Native Application](basic-kotlin-native-app.html#obtaining-the-compiler)
+tutorial for more information on performing this step.
+Let's assume we have a console, where `kotlinc` command is available. 
+
+Now we call the following commands to compile the code into Frameworks
+for macOS, iOS and iOS emulator respectively:
+```bash
+kotlinc lib.kt -produce dynamic -output macOS/Demo
+kotlinc lib.kt -produce framework -target ios_arm64 -output iOS/Demo
+kotlinc lib.kt -produce framework -target ios_x64 -output iOS_emu/Demo
+```
+
+The `kotlinc` generates three frameworks for us, named `Demo.framework` under 
+`macOS`, `iOS` and `iOS_emu` folders respectively.
+
+Let's see what is inside
+
+## Generated Framework Headers
+
+Each of created frameworks contains the header file in `<Framework>/Headers/Demo.h`.
+The file does not depend on the target platform (at least with Kotlin/Native v.0.8.1).
```

Comment:
Not sure if it is correct wrt some primitive type mappings.

Reply:
It should definitely be correct for the code above. But it is not correct when Kotlin code has e.g. `NSInteger` in public signatures.